xla/tests/int4_test.cc

```diff
@@ -186,5 +190,161 @@ XLA_TEST_F(HloTestBaseWithAlgsimpDisabled, TwoDots) {
 )";
   EXPECT_TRUE(RunAndCompare(hlo_text, std::nullopt));
 }
+
+class ElementwiseTest : public HloTestBase,
+                        public ::testing::WithParamInterface<
+                            std::tuple<HloOpcode, PrimitiveType>> {
+ public:
+  static std::vector<HloOpcode> GetOpcodesWithArity(int arity) {
```

Comment:
`GetElementwiseOpcodesWithIntSupportWithArity`?

The name of the function right now is not telling the whole story.

Reply:
Updated the name as you suggest.

Note that the function name usually isn't telling the whole story, the function body does.

Reply:
The function body shouldn't introduce behaviours that are not clear from the name + the docstring. In this case, the behaviour was quite different from what we could expect from just looking at the prototype---that is an antipattern.